src/Features/CSharp/Portable/UseSimpleUsingStatement/UseSimpleUsingStatementDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,134 @@
+                   new LocalizableResourceString(nameof(FeaturesResources.using_statement_can_be_simplified), FeaturesResources.ResourceManager, typeof(FeaturesResources)))
+        {
+        }
+
+        public override bool OpenFileOnly(Workspace workspace) => false;
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+            => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
+
+        protected override void InitializeWorker(AnalysisContext context)
+            => context.RegisterSyntaxNodeAction(AnalyzeSyntax, SyntaxKind.UsingStatement);
+
+        private void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
+        {
+            var usingStatement = (UsingStatementSyntax)context.Node;
+
+            var syntaxTree = context.Node.SyntaxTree;
+            var options = (CSharpParseOptions)syntaxTree.Options;
+            if (options.LanguageVersion < LanguageVersion.CSharp8)
+            {
+                return;
+            }
+
+            if (!(usingStatement.Parent is BlockSyntax parentBlock))
+            {
+                // Don't offer on a using statement that is parented by another using statement.
+                // We'll just offer on the topmost using statement.
+                return;
+            }
+
+            // Check that all the immediately nested usings are convertible as well.  
+            // We don't want take a sequence of nested-using and only convert some of them.
+            for (var current = usingStatement; current != null; current = current.Statement as UsingStatementSyntax)
+            {
+                if (current.Declaration == null)
+                {
+                    return;
+                }
+            }
+
+            // Verify that changing this using-statement into a using-declaration will not
+            // change semantics.
+            if (!PreservesSemantics(parentBlock, usingStatement))
+            {
+                return;
+            }
+
+            var cancellationToken = context.CancellationToken;
+            var optionSet = context.Options.GetDocumentOptionSetAsync(syntaxTree, cancellationToken).GetAwaiter().GetResult();
+            if (optionSet == null)
+            {
+                return;
+            }
+
+            var option = optionSet.GetOption(CSharpCodeStyleOptions.PreferSimpleUsingStatement);
+            if (!option.Value)
+            {
+                return;
+            }
+
+            // Good to go!
+            context.ReportDiagnostic(DiagnosticHelper.Create(
+                Descriptor,
+                usingStatement.UsingKeyword.GetLocation(),
+                option.Notification.Severity,
+                additionalLocations: ImmutableArray.Create(usingStatement.GetLocation()),
+                properties: null));
+        }
+
+        private static bool PreservesSemantics(
```

Comment:
Considering the scenario `using (...) {...} return expr;`. I understand we don't want to offer a "fix", since semantics are changed. 
Should we offer a refactoring though, as the semantic change may be ok for users in some or many cases? (not blocking this PR)

Reply:
Yeah, i've def thought that this could potentially have a refactoring offered, with a potential warning that semantics may change.  but htat could be a future piece of work.